app.ts

```diff
@@ -0,0 +1,216 @@
+window.onload = () => {
+
+    let numColumns: number;
+    let columns: any;
+    let recordCount: number;
+    let startIndex = 0; // Default value = 0;
+    let tableRecordCount = 20; // Default value = 20;
+    let timer: number;
+
+    // Canvas:
+    let canvas = document.createElement("div");
+    canvas.id = "canvas";
+    $("body").append(canvas);
+
+    // Buttons and fields:
+    let searchbtn = document.createElement("button");
+    searchbtn.innerText = "Search";
+
+    let nextbtn = document.createElement("button");
+    nextbtn.innerText = "Next Page";
+
+    let prevbtn = document.createElement("button");
+    prevbtn.innerText = "Previous Page";
+
+    let fromField = document.createElement("input");
+    fromField.placeholder = "from";
+
+    // Window resizing (using a debouncing method):
+    window.onresize = () => {
+        let time = 100;
+        clearInterval(timer);
+        timer = setInterval(function () {
+            clearInterval(timer);
+            adjustTableRecordCount();
+            let end = tableRecordCount - 1;
+            if (recordCount < startIndex + tableRecordCount) {
+                startIndex = recordCount - tableRecordCount;
+            }
+            getRecords(startIndex, startIndex + end);
+        }, time);
+    }
+    
+    // on clicks:
+    searchbtn.onclick = function() {
+        let from = parseInt(fromField.value, 10);
```

Comment:
Behold, an extract from the es5 library mthod for Typescript

```ts
/**
 * Converts a string to an integer.
 * @param s A string to convert into a number.
 * @param radix A value between 2 and 36 that specifies the base of the number in numString.
 * If this argument is not supplied, strings with a prefix of '0x' are considered hexadecimal.
 * All other strings are considered decimal.
 */
declare function parseInt(s: string, radix?: number): number;
```
I include this signature to illustrate that adding the 10 at the end is unnecessary. Please remove.